### DISCUSSION TITLE
Scripts: Expected behavior of multiple {UserInput} in a command

### DISCUSSION THREAD

**Discussion by pmiossec** [gitextensions/gitextensions#11265] [Polls]
Context: using the script feature when a command has multiple `{UserInput}` tokens like for example this one:

`bundle create {UserInput}.bundle -{UserInput} {cLocalBranch}`

![image](https://github.com/gitextensions/gitextensions/assets/460196/059c95bf-95be-4ae8-b933-ade23d2c577d)

I would like your point of view on what is the best/right behavior.

The current behavior is that only one value is requested to the user and it is used to replace all {UserInput} tokens

It seems a little strange to me to be able to request only one value. I would have expected that a value is requested for each {UserInput} token found. Should we fix that? ⚠️  breaking change.

Other intermediate solution:

I have a work in progress to improve a little the `{UserInput}` token allowing to specify a label and a default value that looks like:

`{UserInput:user label|a default value|}`

And we can even use an existing token as (part of ) a default value
example: `{UserInput:Bundle filename|{cLocalBranch}.bundle|}`

So with that improvement, we can think about keeping current behavior when current syntax and replacing only one when specifying the label and default value to make the input kind of unique...

What do you think?

I ask that to know what I should try to fix or not....

**Comment by mstv** [comment]
I use the current behavior for fetching PRs. The extended syntax is welcome.

![image](https://github.com/gitextensions/gitextensions/assets/36601201/9d254156-4a3c-4580-b724-2e0bd2915214)

**Reply by pmiossec** [reply]
Ok.so it is a strange behavior but I understand it is useful and I will keep it.

**Comment by gerhardol** [comment]
Why double curly around? `{{UserInput:Bundle filename|{cLocalBranch}.bundle|}}`

I do not have a situation I see I need this functionality right now myself, so it is not something I push for. But it is potentially useful.

**Reply by pmiossec** [reply]
> Why double curly around? `{{UserInput:Bundle filename|{cLocalBranch}.bundle|}}`

It's an existing feature I just discovered that quote the value returned.
